distributed/s3.py

```diff
@@ -0,0 +1,31 @@
+import boto3
+from dask.imperative import Value
+from distributed import default_executor
+
+DEFAULT_PAGE_LENGTH = 1000
+
+
+def get_list_of_summary_objects(bucket_name, prefix='', delimiter='', page_size=DEFAULT_PAGE_LENGTH):
+    s3 = boto3.resource('s3')
+    return list(s3.Bucket(bucket_name).objects.filter(Prefix=prefix, Delimiter=delimiter).page_size(page_size))
```

Comment:
Ah, I see.  OK, the use of delimiter here conflicts a bit with `distributed.hdfs.read_bytes` where delimiter is used to cut up a large collection into many small collections, ensuring that the cuts are made on delimiter values.  Delimter refers to a bytestring to search for within the bytes being read, not in the path name.

For example, when dealing with text files we use the delimiter `'\n'` to ensure that we always separate blocks on a line boundary.  When reading from avro files we cut on the sixteen byte delimiter specified in the avro header file.  

This only ends up being important if we want to cut a listing into smaller pieces.  I suspect that this will become the case if individual S3 objects grow larger than, say, 1GB. 

We shouldn't necessary resolve this issue now, but it's something to think about.